src/jwk.c

```diff
@@ -1865,11 +1863,19 @@ bool cjose_jwk_hkdf(const EVP_MD *md,
     // HKDF-Extract, HMAC-SHA256(salt, IKM) -> PRK
     unsigned int prk_len;
     unsigned char prk[EVP_MAX_MD_SIZE];
-    HMAC(md, salt, salt_len, ikm, ikm_len, prk, &prk_len);
+    if (NULL == HMAC(md, salt, salt_len, ikm, ikm_len, prk, &prk_len))
```

Comment:
This missing error handling led to some confusion, because the overall cjose_err reported no error while in FIPS mode but the derivation had actually failed.

Reply:
https://www.openssl.org/docs/man1.1.0/man3/HMAC.html

> HMAC() returns a pointer to the message authentication code or NULL if an error occurred.